### PR TITLE
Split the circuits packages into Circom and o1js variants.

### PR DESCRIPTION
**Pull request by SnickerChar**
Left the circom one as is so it won't require any other changes. Didn't publish any new versions, since the Circuits and Circuits packages are effectively unchanged. The new circuits-o1js and circuits-o1js-sdk are unused as yet. Tests pass for all.

### Pre-Flight Checks
- [ ] Has QA approved this change for dev?
- [ ] Are all unit tests passing?
- [ ] Have you added this description to this week's [release notes](https://drive.google.com/drive/folders/1ELnyVZHgISIlwDQXgy0mb-4qsn_1PRZr?usp=sharing)?

### CODE REVIEW

File: packages/circuits-o1js-sdk/test/unit/CommitmentCircuitWrapper.test.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      identityTrapdoor,
+      identityNullifier,
+    );
+  }
+}
+
+describe("CommitmentCircuitWrapper tests", () => {
+  test("Generates Proof", async () => {
+    // Arrange
+    const mocks = new CommitmentWrapperMocks();
```
Comment (OzanEm) [inline packages/circuits-o1js-sdk/test/unit/CommitmentCircuitWrapper.test.ts:29]: It seems like the mock isn't necessary, and there could be room for improvement in the tests. I believe we will work further in the tests

File: packages/circuits-o1js/test/unit/o1jsZkPrograms/Reserves-Circuit.test.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+} from "@circuits-o1js/Reserves.js";
+
+// helper function build generating a bunch of key pairs
+class PublicPrivateKeyMocks {
+  static generateKeys(count: number): [ForeignCurve, CanonicalForeignField][] {
+    const keypairs = new Array<[ForeignCurve, CanonicalForeignField]>();
+
+    for (let i = 0; i < count; i++) {
+      const privateKey = Secp256k1.Scalar.random();
+      const publicKey = Secp256k1.generator.scale(privateKey);
```
Comment (OzanEm) [inline packages/circuits-o1js/test/unit/o1jsZkPrograms/Reserves-Circuit.test.ts:24]: While running these tests, I encountered a TypeError: os.availableParallelism is not a function error. It might be an issue on my end.

**Review by OzanEm — DISMISSED** [review]

File: packages/circuits-o1js/test/unit/o1jsZkPrograms/Reserves-Circuit.test.ts (hunk 2 above)
Comment (SnickerChar) [reply packages/circuits-o1js/test/unit/o1jsZkPrograms/Reserves-Circuit.test.ts:24]: I didn't run into that. Thanks for running the tests though. Let me try again on my side.

**Comment by SnickerChar** [comment]
This is being tested on Demo-03 as a side effect of Ozan's PR

**Review by okanzdmr — DISMISSED** [review]

**Review by orhunerd — DISMISSED** [review]

**Review by OzanEm — APPROVED** [review]

**Review by okanzdmr — APPROVED** [review]

**Review by orhunerd — APPROVED** [review]
